core/templates/list.h

```diff
@@ -522,6 +522,15 @@ class List {
 			it = it->next();
 		}
 	}
+	void operator=(List &&p_list) {
+		if (this == &p_list) {
```

Comment:
You could add the godot equivalent of [[unlikely]] hint there

Reply:
Good idea!